src/ids-base/ids-events-mixin.js

```diff
@@ -18,58 +18,59 @@ const IdsEventsMixin = (superclass) => class extends superclass {
    * @param {Function|any} callback The callback code to execute
    * @param {object} options Additional event settings (passive, once, passive ect)
    */
-  onEvent(eventName, target, callback, options) {
+  onEvent = (eventName, target, callback, options) => {
     target.addEventListener(eventName.split('.')[0], callback, options);
     this.handledEvents.set(eventName, { target, callback, options });
   }
 
   /**
-   * Add and keep track of an event listener.
+   * Remove event listener
    * @param {string} eventName The event name with optional namespace
    * @param {HTMLElement} target The DOM element to register
    * @param {object} options Additional event settings (passive, once, passive ect)
    */
-  offEvent(eventName, target, options) {
+  offEvent = (eventName, target, options) => {
     const handler = this.handledEvents.get(eventName);
+    this.handledEvents.delete(eventName);
     if (handler?.callback) {
       target.removeEventListener(eventName.split('.')[0], handler.callback, options || handler.options);
     }
-    this.handledEvents.delete(eventName);
-  }
+  };
 
   /**
    * Create and trigger a custom event
    * @param {string} eventName The event id with optional namespace
    * @param {HTMLElement} target The DOM element to register
    * @param {object} [options = {}] The custom data to send
    */
-  triggerEvent(eventName, target, options = {}) {
+  triggerEvent = (eventName, target, options = {}) => {
     const event = new CustomEvent(eventName.split('.')[0], options);
     target.dispatchEvent(event);
-  }
+  };
 
   /**
    * Detach all event handlers
    */
-  detachAllEvents() {
+  detachAllEvents = () => {
```

Comment:
```suggestion
  detachAllEvents = () => {
```

Lets keep the original syntax. I think that way it will be more consistent
ets k

Reply:
put a note on why the change in the comments. Basically: on event listener mixins specifically we're always aiming to subscribe to `this` (on Element) but calling the function from other scopes than the object (or even possibly anonymous) was causing issues in some conditions. It's what was causing the test to fail after the small logical change, so that's why I had adjusted.

Reply:
adjusted to use bind in constructor way of doing that